src/serializer/ResidualHeapVisitor.js

```diff
@@ -73,11 +73,12 @@ export class ResidualHeapVisitor {
   modules: Modules;
 
   declarativeEnvironmentRecordsBindings: Map<DeclarativeEnvironmentRecord, VisitedBindings>;
+  // Either realmGenerator or the FunctionValue of an additional function to serialize
   globalBindings: Map<string, VisitedBinding>;
   functionInfos: Map<BabelNodeBlockStatement, FunctionInfo>;
   functionBindings: Map<FunctionValue, VisitedBindings>;
   scope: Scope;
-  realmGenerator: Generator;
+  globalScope: Scope;
```

Comment:
I don't like the name "global scope", as this is confusing in JavaScript. It's not really the global scope, even for the realm.generator. We always emit to a scope nested in an anonymous function.
The word "scope" itself might not be the right word, as JavaScript scopes are weird. Maybe that should be renamed to the more generic word "Context", or "User".